packages/ng/src/app/user/picture/user-picture.component.ts

```diff
@@ -12,19 +12,28 @@ import { LuUserDisplayPipe } from '../display';
 })
 export class LuUserPictureComponent {
 
-	private _user: IUser;
 	/**
 	 * IUser whose picture you wanna display.
 	 */
+	private _user: IUser;
+
+	/**
+	 * User-picture size in px
+	 */
+	@Input() size: number;
+
 	@Input() set user(user: IUser) {
 		this._user = user;
 		this.initials = this.displayPipe.transform(user, 'LF');
 		this.hasPicture = !!user.picture && !!user.picture.href;
+		if (this.size) {
+			this.style = { 'font-size': Math.round(this.size / 2.5) + 'px'};
+		}
 		if (this.hasPicture) {
-			this.style = { 'background-image': `url('${this._user.picture.href}')` };
+			this.style = { ...this.style, 'background-image': `url('${this._user.picture.href}')` };

```

Comment:
the fact you have to use `...this.style` means that you have in mind to add other attributes but others attributes means we are entering in a kind of black box with to many parameters... 

If we really want to go in this way, i prefer to play with a kind of CSS in JS like 

```javascript
{
'font-size':'Math.round(this.size / 2.5) + 'px',
'width' : ....
````

and this a directive for example : `[customStyle]="myObjectWithStyleOverride"`

Reply:
I'm not sure I get your point here. I am using `...this.style` to 'concatenate' the sizing (that is handled by changing the font-size), with the picture image (that can either be a background-image or a background-color if no image is available). It's then used through ngStyle. But it doesn't change the fact that it has its own style defined through a scss file.